src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/TestHost.cs

```diff
@@ -16,74 +16,117 @@
-        }
-    }
+        [OperationContract]
+        [WebGet(UriTemplate = "PeerCert?asPem={asPem}", BodyStyle = WebMessageBodyStyle.Bare)]
+        byte[] PeerCert(bool asPem);
 
-    [ServiceContract]
-    public interface IUtil
-    {
         [OperationContract]
-        byte[] GetClientCert(bool exportAsPem);
+        [WebGet(UriTemplate = "RootCert?asPem={asPem}", BodyStyle = WebMessageBodyStyle.Bare)]
+        Stream RootCert(bool asPem);
 
         [OperationContract]
-        byte[] GetRootCert(bool exportAsPem);
+        [WebGet(UriTemplate = "Fqdn", BodyStyle = WebMessageBodyStyle.Bare)]
+        Stream Fqdn();
 
         [OperationContract]
-        byte[] GetPeerCert(bool exportAsPem);
+        [WebGet(UriTemplate = "Ping", BodyStyle = WebMessageBodyStyle.Bare)]
+        Stream Ping();
 
         [OperationContract]
-        string GetFQDN();
+        [WebGet(UriTemplate = "State", BodyStyle = WebMessageBodyStyle.Bare)]
+        Stream State();
     }
 
-    public class Util : IUtil
+    public class TestHost : ITestHost
     {
-        public byte[] GetClientCert(bool exportAsPem)
+        public Stream ClientCert(bool asPem)
         {
             X509Certificate2 clientCert = CertificateFromSubject(StoreName.My, StoreLocation.LocalMachine, "WCF Client Certificate");
-            if (exportAsPem)
+            byte[] response; 
+
+            if (clientCert == null)
             {
-                return Encoding.ASCII.GetBytes(GetCertificateAsPem(clientCert));
+                WebOperationContext.Current.OutgoingResponse.StatusCode = HttpStatusCode.InternalServerError;
+                response = Encoding.UTF8.GetBytes("Client certificate not found on system");
             }
-            else
+            else 
             {
-                return clientCert.Export(X509ContentType.Pfx, "test");
+                WebOperationContext.Current.OutgoingResponse.Headers["Content-Disposition"] = "attachment; filename=\"wcf-client-cert.crt\"";
+
+                if (asPem)
+                {
+                    WebOperationContext.Current.OutgoingResponse.ContentType = "application/x-pem-file";
+                    response = Encoding.ASCII.GetBytes(GetCertificateAsPem(clientCert));
+                }
+                else
+                {
+                    WebOperationContext.Current.OutgoingResponse.ContentType = "application/x-x509-user-cert";
+                    response = clientCert.Export(X509ContentType.Pfx, "test");
+                }
             }
+
+            return new MemoryStream(response);
         }
 
-        public byte[] GetRootCert(bool exportAsPem)
+        public Stream Crl(string serialNum)
+        {
+            // The test.crl is generated by the cert util tool and will not expire until the root cert expires.
+            // We should investigate if we can generate it in the run time. This is not a blocking issue.
+            string downloadFilePath = @"c:\\WCFTest\\test.crl";
```

Comment:
Ugh :) we use pretty much exactly once (aside from the "DO_NOT_TRUST_WcfBridgeRootCA" literal) 

... which I also want to change to get rid of the "Bridge" reference